5주차 - 레스토랑 조회 구현하기

yxxnhx opened the pull request:
# 목표
레스토랑 조회 구현하기

### 구현사항
- [x] 레스토랑 지역 불러오기
- [x] 레스토랑 카테고리 불러오기
- [x] 레스토랑 목록 불러오기
- [x] 지역, 카테고리 선택 시 체크표시 나오기
- [x] 지역, 카테고리 모두 선택 시 해당 레스토랑 목록 나타내기

### 진행사항 
안녕하세요 트레이너님!
현재 API를 호출까지는 해놓은 상태입니다.
이제 각각의 지역과 목록을 체크하면 그에 맞는 레스토랑이 나올 수 있게 하는 방법에 대하여 고민 중입니다
일단은 차근차근히 해보겠습니다👊

yxxnhx committed:
index 파일 기본 설정

리덕스 사용을 위한 Provider 설정하였습니다.

yxxnhx committed:
store 기본 설정

리덕스 사용을 위한 기본 설정을 하였습니다.

yxxnhx committed:
reducer 기본 설정

리덕스 사용을 위한 reducer 기본 틀을 잡았습니다.

yxxnhx committed:
redux, react-redux, redux thunk 설치

리덕스와 비동기 사용을 위해 리덕스 thunk를 설치하였습니다

yxxnhx committed:
store 미들웨어 thunk 설정

비동기 사용을 위하여 thunk 기본 설정을 해주었습니다.

yxxnhx committed:
App 컴포넌트 dispatch 설정

API 데이터를 불러오기 위해 dispatch를 사용하여 설정하였습니다

yxxnhx committed:
fixtures 폴더 생성 및 테스트용 데이터 생성

테스트용 데이터를 생성하였습니다

yxxnhx committed:
RegionsContainer 생성

지역을 보여줄 RegionsContainer를 생성하였습니다

yxxnhx committed:
RegionsContainer 테스트 파일 생성

지역 목록이 랜더링되는지 여부를 확인할 수 있게 테스트 코드를 생성하였습니다

yxxnhx committed:
상태 관리할 actions 생성

카테고리, 지역, 레스토랑을 관리할 수 있는 액션들을 모아 작업했습니다

yxxnhx committed:
reducer 생성

각 액션별 상태를 관리할 수 있도록 아무런 액션이 없을 경우 기본값을 낼 수 있도록 작업하였습니다

yxxnhx committed:
reducer 테스트 코드 작성

setRegions, setCategories, setRestaurants를 각각 테스트용 데이터를 불러와 제대로 값이 들어오는지 확인하였습니다

yxxnhx committed:
Regions 생성

각 지역을 보여줄 수 있도록 regions를 받아낼 수 있도록 기본 틀을 잡았습니다

yxxnhx committed:
Regions 테스트 코드 구현

지역 목록을 불러와서 받을 수 있도록 테스트코드를 구현했습니다

yxxnhx committed:
CategoriesContainer 생성

카테고리를 Categories 컨테이너에 props를 내려서 화면에 구현할 수 있도록 작업하였습니다

yxxnhx committed:
CategoriesContainer 테스트 코드 생성

카테고리의 목록 랜더링 여부를 테스트 코드로 구현하였습니다

yxxnhx committed:
Categories 컴포넌트 생성

CategoriesContaienr에서 categories를 받아 화면에 보일 수 있도록 틀을 잡았습니다

yxxnhx committed:
Categories 테스트 파일 생성

테스트용 데이터를 받아 제대로 화면에 구현이 되는지 여부를 테스트하였습니다

yxxnhx committed:
api 파일 생성 및 코드 작성

지역, 목록, 레스토랑의 api 데이터를 받아올 수 있도록 async await을 이용하여 구현하였습니다

yxxnhx committed:
mocks api 생성

테스트용으로 사용할 수 있는 빈 api를 생성하였습니다

daadaadaah commented on src/CategoriesContainer.test.jsx:
```diff
@@ -0,0 +1,23 @@
+describe('CategoriesContainer', () => {
+  it('목록이 랜더링된다', () => {
+    useSelector.mockImplementation((selector) => selector({
+      categories: [
+        { id: 1, name: '분식' },
+      ],
```
잘 만들어 놓으신 fixture를 활용해봅시다!ㅎㅎ(다른 테스트에도 fixture를 적용해볼만한 곳이 없을까 한번 체크해보세요~)

yxxnhx replied:
앗 넵! 수정하겠습니다

daadaadaah commented on src/CategoriesContainer.test.jsx:
```diff
@@ -0,0 +1,23 @@
+jest.mock('react-redux');
+
+describe('CategoriesContainer', () => {
+  it('목록이 랜더링된다', () => {
```
어떤 목록인지도 같이 적어주시면, 더 좋을 것 같아요!ㅎㅎ

daadaadaah commented on src/RegionsContainer.test.jsx:
```diff
@@ -0,0 +1,23 @@
+jest.mock('react-redux');
+
+describe('RegionsContainer', () => {
+  it('지역이 랜더링된다', () => {
```
지역 보다 지역 목록 등 복수형이라는 의도를 더 들어내주는게 좋을 것 같아요!ㅎㅎ

yxxnhx replied:
넵 복수형으로 수정하겠습니다!

daadaadaah commented on src/reducer.test.js:
```diff
@@ -0,0 +1,37 @@
+    restaurants: [],
+  };
+
+  describe('setRegions', () => {
+    it('지역을 변경해준다', () => {
+      const state = reducer(initialState, setRegions(regions));
+
+      expect(state.regions).toHaveLength(5);
+    });
+  });
+
+  describe('setCategories', () => {
+    it('카테고리를 변경해준다', () => {
+      const state = reducer(initialState, setCategories(categories));
+
+      expect(state.categories).toHaveLength(5);
+    });
+  });
+
+  describe('setRestaurants', () => {
+    it('레스토랑을 변경해준다', () => {
+      const state = reducer(initialState, setRestaurants(restaurants));
+
+      expect(state.restaurants).toHaveLength(2);
+    });
+  });
```
`목록`이라는 의도를 좀더 드러나게 테스트 내용을 표현해주시는게 좋을 것 같아요!

yxxnhx replied:
넵 의도를 드러낼 수 있도록 목록을 추가하여 수정하겠습니다

daadaadaah reviewed (changes requested):
> 현재 API를 호출까지는 해놓은 상태입니다. 이제 각각의 지역과 목록을 체크하면 그에 맞는 레스토랑이 나올 수 있게 하는 방법에 대하여 고민 중입니다

- 좋습니다! 빠르게 PR 날려주시면서 진행상황까지 코멘트에 남겨주신 점 너무 잘하셨어요👏 
차근차근 하나씩 해봅시당!👊

yxxnhx committed:
CategoriesContainer에 handleClick 이벤트 dispatch 추가

선택시 "V" 표시가 나타날 수 있도록 handleClick 이벤트 추가하였습니다

yxxnhx committed:
CategoriesContainer.test dispatch 호출 여부 테스트 추가

카테고리를 클릭 시  dispatch가 setCategory와 함께 호출이 되는지 여부를 테스트 추가하였습니다

yxxnhx committed:
RegionsContainer handleClick 이벤트 dispatch 추가

지역 목록 중 하나를 선택했을 때, V 표시를 랜더링하기 위하여 handleClick 이벤트를 추가하였습니다

yxxnhx committed:
RegionsContainer.test dispatch 호출 여부 테스트 추가

지역을 선택하면 distpatch가 selectRegion과 함께 호출되는지 여부 확인하는 테스트를 추가하였습니다

yxxnhx committed:
action selectCategory, selectRegion 추가

각각 선택한  지역, 카테고리를 받아올 수 있는 액션을 추가하였습니다.

yxxnhx committed:
reducer selectCategory, selectRegion 추가

선택한 지역, 카테고리를 받아올 수 있는 액션을 reducer에 추가하였습니다

yxxnhx committed:
Categories에 선택 여부, 목록 여부 추가

선택이 되었을 경우 카테고리의 아이디와 선택한 카테고리의 아이디를 동일하게 반환하는 조건과 아무것도 데이터가 들어오지 않았을 때의 조건을 추가하였습니다.

yxxnhx committed:
Categories test 목록 랜더링 여부, handleClick 호출, 아이디 여부 테스트 코드 추가

카테고리 목록 데이터 여부, handleClick 호출 여부, 아이디 존재 여부에 따른 테스트 코드를 추가하였습니다

yxxnhx committed:
Region 컴포넌트 생성 및 구현

기존 Regions에서 각각의 li들을 따로 분리하여 삼항연산자를 사용하여 선택되었을 시에만 "V" 표시가 나타나도록 구현하였습니다

yxxnhx committed:
Category test 파일 생성 및 코드 추가

각각의 카테고리 목록이 랜더링 여부, handleClick 여부, 선택시 "V" 표시 추가 여부를 확인하는 테스트 코드를 생성하였습니다

yxxnhx committed:
Regions에 선택 여부, 목록 여부 추가

선택된 지역의 아이디 여부, 지역 목록 데이터 여부를 확인할 수 있는 조건을 추가하였습니다

yxxnhx committed:
Regions  test 파일 생성 및 코드 추가

데이터 목록 여부, handleClick 여부, 아이디 존재 여부를 확인할 수 있는 테스트 코드를 생성하였습니다

yxxnhx committed:
Region 컴포넌트 생성 및 코드 추가

기존의 Regions에 있던 각각의 지역들을 Region에 분리하여 삼항연산자를 사용하여 선택되었을 시 "V"가 나타나도록 하였습니다

yxxnhx committed:
Region.test 생성 및 테스트 코드 구현

지역 목록 여부, handleClick 호출, 아이디 여부에 따른 테스트를 추가하였습니다

yxxnhx committed:
reducer 목록 의도에 따른 테스트 코드 수정

명시성을 위하여 카테고리, 지역, 레스토랑 목록을 변경하도록 테스트 조건 수정하였습니다

yxxnhx commented:
# 진행사항
현재 지역과 카테고리를 선택 시 (V)까지 나타낼 수 있도록 하였습니다.
이제 이 선택된 지역과 카테고리를 받아서 레스토랑 목록이 보일 수 있도록 진행중입니다.

## 질문사항
``` jsx
const isSelected = (category) => {
    if (isSelected === null) {
      return false;
    }

    return category.id === selectedCategoryId;
  };
```
현재 선택한 카테고리의 id를 카테고리의 id로 넘기고 만약 선택한 것이 아닐 때는 false를 반환하도록 하였습니다.

``` jsx
  context('카테고리 아이디가 없을 시', () => {
    it('"V"가 랜더링되지 않는다', () => {
      const category = categoryList[0];
      const { container } = renderCategories({
        categories: category,
        selectedCategoryId: null,
      });

      expect(container).not.toHaveTextContent('(V)');
    });
  });
```
그래서 아이디가 없을 시에는 (V)가 없다고 테스트를 하였는데 테스트 커버리지에서 해당 부분이 테스트가 되지 않았다고 합니다
그렇다면 말 그대로 테스트 코드에 toBeFalsy가 있어야 하는건가요?
아니면 다른 방법이 있을까요?😢

daadaadaah commented on src/Categories.jsx:
```diff
@@ -0,0 +1,30 @@
+
+export default function Categories({ categories, selectedCategoryId, onClick }) {
+  const isSelected = (category) => {
+    if (isSelected === null) {
```
`selectedCategoryId` 상태를 활용해서 선택 유무 판별할 수 있을 것 같은데, isSelected라는 함수로 빼신 이유가 궁금합니다!

yxxnhx replied:
선택된 것을 조금 더 명확하게 해주고 싶어서 따로 빼서 확인하는 함수를 만들어서 사용을 했는데 굳이 필요가 없다면 그냥 `isSelected` 자리에 그대로 `selectedCategoryId`를 넣어서 사용하는 것이 더 옳은 것일까요?

daadaadaah replied:
아니요! 그냥 의도가 궁금했어요! 그런데, `selectedCategoryId`로 넘겨주고,  만드신 함수를 Category에서 만들어서 사용해주면, Categories와 Category 각각 의도가 좀더 드러나지 않을까 하는데, 어떻게 생각하시나여?

daadaadaah commented on src/Categories.test.jsx:
```diff
@@ -0,0 +1,71 @@
+    />
+  ));
+
+  context('카테고리 데이터가 없을 시', () => {
```
`데이터` 라는 용어를 빼도 의도 전달에 문제가 없을 것 같아요! 대신 카테고리 목록이라 표현해주시는건 어떠신가여?ㅎㅎ

yxxnhx replied:
아 그렇군요! 데이터 대신 목록으로 수정하겠습니다~!

daadaadaah commented on src/Categories.test.jsx:
```diff
@@ -0,0 +1,71 @@
+    });
+  });
+
+  context('카테고리 아이디가 있을 시', () => {
```
`선택한 카테고리 아이디가 잇을 시` 라고 표현하는게 좀더 적절해보이는데, 어떻게 생각하세여?ㅎㅎ

yxxnhx replied:
아 그렇네요 카테고리 전체는 다 아이디가 있으니까요ㅠㅠ 수정하겠습니다!

daadaadaah commented on src/CategoriesContainer.test.jsx:
```diff
@@ -0,0 +1,44 @@
+    expect(screen.getByText('분식')).not.toBeNull();
+  });
+
+  context('카테고리를 클릭 시', () => {
```
context는 대비되는 상황에서 사용하는게 적절해요! 지금은 대비되는 상황이 없으니 그냥 it문에 표현해주셔도 될 것 같아요!ㅎㅎ

yxxnhx replied:
아하 그렇군요! 수정하겠습니다~!

daadaadaah commented on src/Category.test.jsx:
```diff
@@ -0,0 +1,58 @@
+    expect(handleClick).toBeCalledWith(category.id);
+  });
+
+  context('카테고리를 선택하면', () => {
```
지금 현재 props로 내려준 상태에 대해 테스트 하는 것이므로, 테스트 코드에 선택한다는 의미가 담긴 표현이 없습니다. 그래서, `선택된 카테고리가 있으면` 이 좀더 적절하지 않을까 하는데, 어떻게 생각하시나요?

yxxnhx replied:
그렇네요! 선택이 되었다는 표현이 조금 더 정확한 것 같아요!

daadaadaah commented on src/Regions.test.jsx:
```diff
@@ -0,0 +1,69 @@
+    />
+  ));
+
+  context('카테고리 데이터가 없을 시', () => {
```
it문이 빠졌어요!

yxxnhx replied:
아이고.. 넵 바로 수정하겠습니다~!

daadaadaah reviewed (changes requested):
> 테스트 커버리지에서 해당 부분이 테스트가 되지 않았다고 합니다

테스트 커버리지 상태를 같이 보여주시면 답변에 도움될 것 같아요!ㅎㅎ

yxxnhx committed:
RestaurantsContaienr 컴포넌트 생성 및 구현

레스토랑 목록이 보일 수 있게 레스토랑 관련 관심사를 분리하여 생성하였습니다.
선택한 지역과, 카테고리 아이디가 있을 시에만 loadRestaurnats를 호출하여 상태를 변화시킬 수 있도록 하였습니다.

yxxnhx committed:
RestaurantsContainer.test 테스트 파일 생성 및 테스트

선택한 지역과 카테고리 데이터 여부를 나누어 랜더링 되는 것을 확인하는 테스트 코드를 구성하였습니다

yxxnhx committed:
Restaurants 컴포넌트 생성 및 코드 작성

RestaurantsContainer에서 restaurants를 props 받아서 화면에 보일 수 있도록 구성하였습니다

yxxnhx committed:
Restaurants.test 테스트 파일 생성 및 구현

레스토랑 데이터 여부에 따른 테스트 코드를 구현하였습니다

yxxnhx committed:
Restaurant 컴포넌트 생성 및 구현

Restaurants에서 restaurant를 받아 이름을 보여줄 수 있도록 관심사를 분리하여 생성하였습니다

yxxnhx committed:
Restaurant.test 테스트 파일 생성 및 구현

레스토랑 목록이 랜더링되는 것을 확인하는 테스트 코드를 구성하였습니다

yxxnhx committed:
Region, Regions, RegionsContainer의 selectedRegionId에서 selelctedRegiond으로 의도가 드러나게 변경

레스토랑이 보일 수 있게 넘기는 데이터가 Regoin의 id가 아닌 region이어서 의도에 맞게 변수명 변경하였습니다

yxxnhx committed:
Region.test, Regions.test, RegionsContainer.test 의도에 맞게 테스트 코드 수정

이전에 카테고리로 잘못 들어간 부분들을 지역으로 변경하였습니다

yxxnhx committed:
data에 테스트용 레스토랑 데이터 변경

들어오는 데이터를 확인하기 위하여 조금 더 세분화하여 변경하였습니다

yxxnhx committed:
reducer에 초기값 selecteRegion, selectedCategoryId 추가

선택한 값들의 상태를 변경하기 위하여 초기값으로 빈값을 추가해주었습니다

yxxnhx committed:
actions의 loadRestaurants 수정

이전에 간단히 틀만 잡은 것에서 selectedRegion과 selectedCategoryId가 들어와서 fetchRestaurnts에 넘길 수 있도록 수정하였습니다

yxxnhx committed:
App 컴포넌트에 RestaurntsContainer 추가

레스토랑 목록이 보일 수 있도록 추가해주었습니다

yxxnhx committed:
api의 fetchRestaurants 수정

템플릿 리터럴로 변경하여 변수를 받아 사용할 수 있도록 수정하였습니다

yxxnhx committed:
reducer.test에 selectCategory, selectRegion 추가

선택한 카테고리와 지역으로 변경을 해주는 액션을 테스트 추가하였습니다

yxxnhx committed:
Categories 테스트 조건 의도에 맞게 수정

데이터 대신 목록, 선택한 카테고리로 수정하여 테스트의 의도를 드러내었습니다

yxxnhx committed:
CategoriesContainer 테스트 코드 상황에 맞게 수정

대비 상황이 아닌 것은 it으로 수정하였습니다

yxxnhx committed:
Category 테스트 조건 상황에 맞게 수정

props를 받은 컴포넌트이므로 선택한에서 선택된으로 수정하였습니다

yxxnhx committed:
Region 테스트 조건 상황에 맞게 소정

props를 받은 컴포넌트 이므로 조건을  수동형으로 수정하였습니다

yxxnhx committed:
Regions 테스트 파일 it 추가

테스트 파일에서 상황 제시만 하고 테스트를 하지 않은 부분 추가하였습니다

yxxnhx commented:
> > 테스트 커버리지에서 해당 부분이 테스트가 되지 않았다고 합니다
> 
> 테스트 커버리지 상태를 같이 보여주시면 답변에 도움될 것 같아요!ㅎㅎ

yxxnhx committed:
RestaurnatsContainer 테스트 파일 useSelector 추가

레스토랑 목록이 비어있을 때의 조건에 맞게 빈 값으로 useSelector를 추가하였습니다

yxxnhx commented:
# 진행사항
레스토랑 목록을 불러와서 해당하는 레스토랑 목록이 화면에 나타날 수 있도록 하였습니다
현재 테스트 코드를 작성 중입니다.

### 스크린샷
![Nov-11-2022 17-25-22](https://user-images.githubusercontent.com/50559373/201298041-01756f8e-0e60-4699-97db-41c436599f0a.gif)

### 질문사항
<img width="609" alt="image" src="https://user-images.githubusercontent.com/50559373/201298761-7f7dfdb2-5f91-49c2-bc53-a1848335fce6.png">

테스트 커버러리지에서 20번째 줄, 아무것도 반환하지 않는 부분의 테스트를 위하여

<img width="703" alt="image" src="https://user-images.githubusercontent.com/50559373/201298451-159933fe-43d0-414f-a88f-ffb6c5173524.png">

위와 같이 RestaurantsContainer에서 선택된 카테고리나 지역이 없을 시에는 dispatch가 실행되지 않은 부분을 테스트를 추가하려 위와 같이 작성을 하면 이미 함수가 호출되었다고 계속 에러가 납니다
왜 그런 것일까요?ㅠㅠ


그리고 actions의 loadRestaurnts 혹은 api의 fetch를 테스트를 하기 위해서 하다가 계속 막혀서 다른 분들이 것을 참고하려 보니 다른 라이브러리를 사용해서 store을 mock으로 사용할 수 있게 하시던데 다른 라이브러리를 꼭 깔아야만 하는건가요?

yxxnhx committed:
api.test 테스트 파일 생성 및 구현

각각의 fetchRegions, fetchCategories, fetchRestaurants의 url을 가져오는지, 목록을 가져오는지 여부를 테스트하였습니다

yxxnhx committed:
RestaurantsContainer.test 테스트 조건 수정

상황에 맞게 선택한 지역 혹은 카테고리 아이디가 없을 시에는 디스패치 대신 loadRestaurants가 실행되지 않는다는 것으로 조건을 수정하였습니다

yxxnhx committed:
actions.test 테스트 파일 생성 및 구현

loadCategories, loadRegions, loadRestaurants를 각각 필요한 디스패치를 호출하는 여부를 테스트할 수 있도록 구현하였습니다

yxxnhx commented:
# 진행사항
actions에서 loadRegions, loadCategories, loadRestaurants까지 테스트 마쳐 테스트 커버리지까지 통과한 상태입니다
위에서 질문한 dispatch가 호출되지 않는다 대신 loadRestaurants가 호출되지 않는다로 조건을 수정하여 코드를 정리하였습니다

### 질문사항
``` jsx
  useEffect(() => {
    if (selectedRegion === null || selectedCategoryId === null) {
      return;
    }

    dispatch(loadRestaurants(selectedRegion, selectedCategoryId));
  }, [selectedRegion, selectedCategoryId]);
```
그런데 제가 생각하기에는 selectedRegion, selectedCategoryId가 모두 null이면 dispatch가 호출되지 않을 것이라고 생각하였는데 호출을 하는 것일까요?ㅠㅠ

daadaadaah commented on src/Categories.jsx:
```diff
@@ -0,0 +1,30 @@
+          <Category
+            key={category.id}
+            category={category}
+            isSelected={isSelected(category)}
```
props명과 함수명이 같으면 혼동스러울 것 같아요!

yxxnhx replied:
앗 그렇겠네요 반영하여 수정하겠습니다!

daadaadaah commented on src/RestaurantsContainer.test.jsx:
```diff
@@ -0,0 +1,72 @@
+    it('loadRestaurants가 실행되지 않는다', () => {
+      useSelector.mockImplementation((state) => state({
+        restaurants: [],
+        selectedRegion: null,
+        selectedCategory: null,
```
`selectedCategoryId `랑 `selectedCategory`랑 다른 값을 다루는 변수인가요?

yxxnhx replied:
앗 제가 변수명을 통일시키지 못했네요 수정하겠습니다

daadaadaah reviewed (changes requested):
> 그런데 제가 생각하기에는 selectedRegion, selectedCategoryId가 모두 null이면 dispatch가 호출되지 않을 것이라고 생각하였는데 호출을 하는 것일까요?ㅠㅠ

1. 일단 실제 코드의 `selectedCategoryId`이랑 테스트 코드의 `selectedCategory` 먼저 체크해보시는 걸 추천 드립니다.
2. 실제로 호출이 되는지, 그럼 왜 호출이 되는지를 알기 위해 각각의 데이터를 console로 찍어보셨나요? 
3. 추가로, `beforeEach`와 `jest.clearAllMocks();`에 대해서 공부해보세요!

yxxnhx committed:
Categories 컨포넌트 관심사 분리

기존에 선택된 카테고리 여부를 확인할 수 있는 것을 Category에서 전체적으로 관리할 수 있도록  Categories에서는 카테고리 목록만 넘길 수 있게 관심사 분리를 하였습니다

yxxnhx committed:
Category 관심사 분리

선택된 카테고리를 바로 Category에서 처리할 수 있도록 selectedCategoryId를 받아서 기존 category의 id가 같을 경우 바로 체크 표시가 나타날 수 있도록 변경하였습니다

yxxnhx committed:
Regions 컴포넌트 관심사 분리

선택된 지역을 관리하는 것은 Region 컴포넌트에서 전체적으로 관리할 수 있도록 Regions에서는 목록만 넘길 수 있게 관심사 분리를 하였습니다

yxxnhx committed:
Region 컴포넌트 관심사 분리

selectedReigon과 region의 이름이 같을 경우 바로 체크표시가 나타날 수 있도록 관심사 분리를 하여 수정하였습니다

yxxnhx committed:
RestaurantsContainer 컴포넌트의 mock 초기화 설정

아무런 데이터가 없을 경우 dispatch가 실행되지 않음을 테스트하기 위하여 mock 초기화 설정을 추가하여 테스트를 통과시켰습니다.

yxxnhx committed:
Category 테스트 파일 수정

기존에 isSelected 함수를 받아 체크 표시 여부를 확인하였던 것에서 Category 컴포넌트 자체적으로 확인하는 조건으로 변경되어 테스트 코드를 selectedCategoryId가 없을 경우로 변경하여 테스트 코드를 수정하였습니다

yxxnhx committed:
Region 테스트 파일 조건 수정

기존에 isSelected 함수를 받아 체크 표시 여부를 확인하였던 것에서 Region 컴포넌트 자체적으로 확인하는 조건으로 변경되어 테스트 코드를 selectedRegion이 없을 경우로 변경하여 테스트 코드를 수정하였습니다

yxxnhx committed:
지역을 선택하지 않고도 분류를 선택 시 해당 카테고리의 식당이 나오는 에러 수정

그 에러를 수정하기 위해 selectedRegion, selectedCategoryId의 기본값을 null에서 빈 값으로 수정하였습니다

yxxnhx commented:
# 진행사항
`beforeEach`와 `jest.clearAllMocks();`를 활용하여 dispatch가 호출되지 않는 조건으로 일어났던 에러를 해결하였습니다!
그리고 트레이너님 조언을 토대로 Categories와 Regions에서는 각각의 목록 데이터만, Category, Region에서는 데이터를 받아 선택하는 여부를 관리할 수 있게 관심사를 분리하여 그에 맞게 테스트 파일의 조건도 수정해주었습니다.
그리고 이전에 `selectedRegion`, `selectedCategoryId`를 기본값을 `null`을 주었을 때 지역을 선택하지 않고 카테고리만 선택하였을 때 카테고리에 해당하는 식당 목록이 보이는 에러를 잡기 위해 `null`에서 빈값 `''`으로 수정하여 에러를 수정하였습니다.

daadaadaah reviewed (approved):
이번주도 수고 많으셨습니다!ㅎㅎ다음주도 같이 힘내봐요!👊
